TLS 1.3: Rename early_data and max_early_data_size configuration function (#8487)
Mbed-TLS/mbedtls: ChangeLog.d/rename-conf-early-data-API.txt

> API changes
>    * Remove `tls13_` prefix in the name of mbedtls_ssl_tls13_conf_early_data()
>      and mbedtls_ssl_tls13_conf_max_early_data_size(). Early data feature may
>      not be TLS 1.3 specific in the furture, as requested in #6909.

Reviewer: ```suggestion
API changes
   * Remove `tls13_` in mbedtls_ssl_tls13_conf_early_data() and
     mbedtls_ssl_tls13_conf_max_early_data_size() API names. Early data
     feature may not be TLS 1.3 specific in the future. Fixes #6909.
```

API changes

>    * In TLS 1.3 early data, rename mbedtls_ssl_tls13_conf_early_data() and
>      mbedtls_ssl_tls13_conf_max_early_data_size() to
>      mbedtls_ssl_conf_early_data() and mbedtls_ssl_conf_max_early_data_size()
>      respectively.

Reviewer 2: ```suggestion
   * Remove `tls13_` prefix in the name of  `mbedtls_ssl_tls13_conf_early_data()` and `mbedtls_ssl_tls13_conf_max_early_data_size()` . Early data feature may not be TLS 1.3 specific in the furture.
```
